### PR TITLE
(esp_modem): Fix switching to CMUX mode for devices that send `MSC`s

### PR DESCRIPTION
**Pull request by david-cermak**
fix(esp_modem): CMUX to ignore MSC frames
    
    Otherwise it gets confused with DISC frame and causes trouble when entering
    CMUX mode (if device sends MSC requests)

### CODE REVIEW

**Comment by david-cermak** [comment]
@RaD PTAL, as well. (have only removed the delay for the control terminal, as I think it's not needed)

**Comment by RaD** [comment]
@david-cermak should I do something to help with this merge request?

**Comment by david-cermak** [comment]
> @david-cermak should I do something to help with this merge request?

You can go through the code changes proposed in this PR if you feel like reviewing it (note that anyone can help with reviews and approve PRs on GitHub, and this repo requires at least one approval for a PR to be merged)

**Review by RaD — APPROVED** [review]
Works well

**Comment by david-cermak** [comment]
> Works well

Thanks for the review, I've added [52efbbe](https://github.com/espressif/esp-protocols/pull/142/commits/52efbbe5383686e001b077952e8b5852c6ebe026) to have the delay configurable per the discussion in https://github.com/espressif/esp-protocols/issues/140#issuecomment-1257708278

File: components/esp_modem/Kconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+menu "esp-modem"
+
+    config ESP_MODEM_CMUX_DEFRAGMENT_PAYLOAD
+        bool "Defragment CMUX messages internally"
+        default y
+        help
+            If enabled (default), the esp-modem internally defragments CMUX messages
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply components/esp_modem/Kconfig]: @euripedesrocha Could you please check if it looks okay to you now?

**Review by euripedesrocha — APPROVED** [review]

**Pull request merged**